src/inmanta/server/services/compilerservice.py

```diff
@@ -489,9 +517,9 @@ async def get_report(self, compile_id: uuid.UUID) -> Apireturn:
         return 200, {"report": report}
 
     @protocol.handle(methods.get_compile_queue, env="tid")
-    async def get_compile_queue(self, env: data.Environment) -> List[CompileRun]:
+    async def get_compile_queue(self, env: data.Environment) -> List[model.CompileRun]:
         """
             Get the current compiler queue on the server
         """
         compiles = await data.Compile.get_next_compiles_for_environment(env.id)
-        return [x.to_dto() for x in compiles]
+        return [x.to_dto() for x in unique_everseen(compiles, CompilerService._compile_merge_key)]
```

Comment:
I don't think it is a good idea to compact the view on the queue as well. 

This would mean that if you queue a compile and you request the queue, it may not be in the queue even if it is not done yet.

Reply:
I found this a hard decision to make. I think what made me decide to compact it here as well is the way Jenkins jobs are queued and the fact that that feels very intuitive. But this is of course not entirely the same thing so I do understand your perspective. I'll revert this to its original implementation.